### ISSUE TITLE
Checklist does not load

### ISSUE THREAD

**Issue by brimwats** [delashum/obsidian-checklist-plugin#42]
I've tried uninstalling/reinstalling/deleting data but all the checklist does is sticks on "loading" mode 
![image](https://user-images.githubusercontent.com/49103046/112997916-65564400-9122-11eb-9420-435940d2cf6b.png)

**Comment by delashum** [comment]
Can you screenshot me the error the console is throwing?

**Comment by brimwats** [comment]
![image](https://user-images.githubusercontent.com/49103046/113027062-87f65600-913e-11eb-9a71-b45f6aae327c.png)

**Comment by matpatterson** [comment]
I am seeing this same issue

**Comment by delashum** [comment]
Okay yeah this is the same issue as #33. The problem is that there are multiple ways to format a YAML list:

```yaml
# A
tags: [one, two]
```
```yaml
# B
tags: one, two
```
```yaml
# C
tags: 
  - one
  - two
```

and I only support `A`. So either this plugin needs to support all of these formats, or you can change your YAML to use `A`.

I am curious, how do you have your tags formatted?

**Comment by brimwats** [comment]
Ah that means that it would be incompatible with the latest changes in the mdnotes extension changes which does YAML like C
https://forum.obsidian.md/t/zotero-zotfile-mdnotes-obsidian-dataview-workflow/15536 

I hadn't used YAML or formatted my tags until this week, I wanted to use checklist with mdnotes in order to auto extract certain highlights as checklists that I could come back to

**Comment by delashum** [comment]
Yeah, and they are adding the hash before it as well, which not all do. This is a tricky problem since everyone seems to format their YAML a bit differently. I will just add support for all these then

**Comment by Chetachiiii** [comment]
I'm also having the same issue. I used the plugin before and it worked fine. My format has always been the same.
<br>

<img width="1680" alt="Screen Shot 2021-04-03 at 3 01 27 PM" src="https://user-images.githubusercontent.com/79069364/113490641-134e4f00-9491-11eb-95f3-65a9bd88c115.png">


<img width="552" alt="Screen Shot 2021-04-03 at 3 30 09 PM" src="https://user-images.githubusercontent.com/79069364/113490706-7f30b780-9491-11eb-8371-711a62ba5fdb.png">

**Comment by pjeby** [comment]
Perhaps the plugin should use the `parseFrontMatterTags()` API?

```typescript
import {parseFrontMatterTags} from "obsidian"

// ...

const getFrontmatterTags = (cache: CachedMetadata, todoTag?: string) => {
    const frontMatterTags: string[] = parseFrontMatterTags(cache?.frontmatter) ?? [];
    // ...
}
```

This will handle tags that are formatted as "x, y, z" or "x" as well as the correct "[x, y, z]".  It also handles fields named `Tags`, `tAg` or any case/plurality variant thereof, making your parsing compatible with how Obsidian does it.  The returned tag list also has the `#` added, if it was omitted from the frontmatter...  but doesn't double them up if they were already there.

**Comment by delashum** [comment]
Well I'll be damned, I had no idea that was part of the API. That's definitely the way to do it. Do you know if supports method `C` that i described above?

**Comment by pjeby** [comment]
Your case C is the same as case A, in the sense that it is a YAML list value.  Case B is a string value that contains commas.  The API handles both lists and comma-delimited strings, so yes.

The type declarations for front matter parsing functions can be found here: https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts#L1894

**Comment by delashum** [comment]
fixed in `1.2.0`
